aws-cpp-sdk-core/include/aws/core/utils/memory/stl/AWSMap.h

```diff
@@ -13,6 +13,21 @@
 #include <unordered_map>
 #include <cstring>
 
+#ifdef USE_AWS_MEMORY_MANAGEMENT
+#include <aws/core/utils/HashingUtils.h>
+/* Inject hash method for an Aws::String with a custom allocator to workaround original C++ defect:
+ * "hash support for std::basic_string with customized allocators was not enabled"
+ * see LWG 3705: https://en.cppreference.com/w/cpp/string/basic_string/hash */
+namespace std
+{
+    template<>
+    struct std::hash<Aws::String>
+    {
+        std::size_t operator()(const Aws::String& t) const { return Aws::Utils::HashingUtils::HashString(t.c_str()); }
```

Comment:
I wish I could just redirect to a std::basic_string::hash, but inner implementation details are not accessible outside regular std::string (which is not compatible with our custom-allocator string)